reconcile/ocm_addons_upgrade_scheduler_org.py

```diff
@@ -35,12 +36,25 @@ def calculate_diff(
     return diffs
 
 
-def run(dry_run: bool) -> None:
-    # patch integration name for state usage
-    ous.QONTRACT_INTEGRATION = QONTRACT_INTEGRATION
-    settings = queries.get_app_interface_settings()
-    ocms = queries.get_openshift_cluster_managers()
+@dataclasses.dataclass(unsafe_hash=True)
+class ResultKey:
+    ocm_map: OCMMap
+    ocm_name: str
+    addon_id: str
+
+
+@dataclasses.dataclass
+class Result:
+    current_state: list[dict[str, Any]]
+    desired_state: list[dict[str, Any]]
+    diffs: list[dict[str, str]]
+
+
+def compute(
```

Comment:
The `current_state`, `desired_state` and `diffs` computation were basically extracted from the `run` function into this `compute` function. This allows to reuse it easily in the qontract-cli command.
The `compute` function returns a `Result` for each org/addon (= `ResultKey`)